commit 27e85d7199ea979780210371f0c0b4f1279f817b (refs/changes/46/1346/3)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2017-01-26T12:57:48-08:00 (2 years, 8 months ago)
    
    CNAPI-690 CNAPI needs to be selective with logging request payload

diff --git a/lib/server.js b/lib/server.js
index 7276d2c..a6137c9 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -13,6 +13,7 @@ var restify = require('restify');
 var restifyValidator = require('restify-validator');
 var trace_event = require('trace-event');
 var formatJSON = require('restify/lib/formatters/json');
+var jsprim = require('jsprim');
 
 var endpoints = require('./endpoints/index');
 var request_seq_id = 0;
@@ -32,6 +33,7 @@ function createServer(options) {
         'servereventheartbeat': true,
         'servereventvmsupdate': true
     };
+
     cnapi.use(function setTracing(req, res, next) {
         req.trace = trace_event.createBunyanTracer({
             log: req.log
@@ -72,18 +74,28 @@ function createServer(options) {
         'servereventheartbeat': true,
         'servereventvmsupdate': true
     };
+
+    var AUDIT_OMIT_BODY_ROUTES = {
+        'vmdockerbuild': true
+    };
+
     cnapi.on('after', function (req, res, route, err) {
         if (req.route && AUDIT_SKIP_ROUTES[req.route.name]) {
             return;
         }
 
         // Successful GET res bodies are uninteresting and *big*.
-        var body = req.method !== 'GET' &&
-                   res.statusCode !== 404 &&
-                   Math.floor(res.statusCode/100) !== 2;
+        var isOmitBodyRoute = (jsprim.hasKey(AUDIT_OMIT_BODY_ROUTES,
+                                         req.route.name) &&
+                                 AUDIT_OMIT_BODY_ROUTES[req.route.name]);
+
+        var includeBody = !isOmitBodyRoute && (req.method !== 'GET' &&
+                        res.statusCode !== 404 &&
+                        Math.floor(res.statusCode/100) !== 2);
+
         restify.auditLogger({
             log: req.log.child({ route: route && route.name }, true),
-            body: body
+            body: includeBody
         })(req, res, route, err);
     });
 
